scripts/lib/upgrade-zulip-stage-2

```diff
@@ -89,7 +89,10 @@ deploy_path = args.deploy_path
 os.chdir(deploy_path)
 
 config_file = get_config_file()
-tornado_processes = int(get_config(config_file, 'application_server', 'tornado_processes', '1'))
+if config_file.has_section('tornado_sharding'):
+    tornado_processes = max(1, len(config_file.options('tornado_sharding')))
+else:
+    tornado_processes = 1
```

Comment:
I guess one thing that isn't great about this block is that if we add anything that's not of the form `port-`, it'll be a bit fragile.  We're also repeating this logic roughly 4 times.

Maybe we want to take the block in `restart-server` and put in a shared zulip_tools.py function returning the list of Tornado ports, and then logic like this can just do `len()` that list?

Reply:
> I guess one thing that isn't great about this block is that if we add anything that's not of the form `port-`, it'll be a bit fragile.

Not sure what you mean here, particularly with `port-`.  You mean that it constrains the `[tornado_sharding]` config section to only contain port numbers as option names?  That doesn't seem too obious a restriction; and certainly refactoring it into a common block (see below) makes it easier to filter for just numbers, in the future.

> We're also repeating this logic roughly 4 times. [...] Maybe we want to take the block in `restart-server` and put in a shared zulip_tools.py function returning the list of Tornado ports, and then logic like this can just do `len()` that list?

That's a great place to combine the logic for most of them!  I was still thinking of all of these as standalone, and lamenting that there was no place to dedup it into.  I do feel a little odd about adding a dependency from `zproject/computed_settings.py` to `scripts/lib/zulip_tools.py`, though.

Reply:
> ... config section to only contain port numbers

Yeah, that's the concern.  And I agree we can resolve it; if nothing else, we can just modify this code to only look at things starting with a number.

We already depend on `zulip_tools.py` from the settings files:

```
tabbott@coset:~/zulip$ git grep zulip_tools zproject/
zproject/default_settings.py:from scripts.lib.zulip_tools import deport
zproject/dev_settings.py:from scripts.lib.zulip_tools import deport
```

The purpose of that library is to be the one thing that can be imported from anywhere on the server, and this seems like a good fit for something to include there.